tests/admin/views/test_rules.py

```diff
@@ -2238,6 +2238,124 @@ def testAddScheduledChangeMultipleConditions(self):
         ret = self._post("scheduled_changes/rules", data=data)
         self.assertEqual(ret.status_code, 400)
 
+    @mock.patch("time.time", mock.MagicMock(return_value=300))
+    def testAddScheduledChangesRuleWithDuplicateAliasWithChangeTypeInsert(self):
+        ret = self._post(
+            "/rules", data=dict(backgroundRate=31, mapping="c", priority=33, product="Firefox", update_type="minor", channel="nightly", alias="test")
+        )
+        self.assertEqual(ret.status_code, 200, "Status Code: %d, Data: %s" % (ret.status_code, ret.get_data()))
+
+        data1 = {
+            "when": 1234567,
+            "priority": 120,
+            "backgroundRate": 100,
+            "product": "blah",
+            "channel": "blah",
+            "mapping": "a",
+            "change_type": "insert",
+            "base_alias": "test",
+        }
+        ret1 = self._post("/scheduled_changes/rules", data=data1)
```

Comment:
This is returning a 400 even if I comment out the previous post:
> b'{"type": "about:blank", "title": "Bad Request", "detail": "Null/Empty Value", "status": 400, "exception": "update_type cannot be set to null/empty when scheduling insertion of a new rule"}'

This suggests that this test is not failing for the right reason. Please add an additional assert about the `exception` message from `ret1.get_data()` to make sure this fails for the right reason, and fix the test to do so. (Please do this for the other tests as well -- they're probably hitting the same issue.

Reply:
Ok got it must be I'll update the tests